contracts/service-registry/src/state.rs

```diff
@@ -154,6 +173,31 @@ pub fn save_new_service(
         .into_report()
 }
 
+pub fn update_authorized_verifier_count(
+    storage: &mut dyn Storage,
+    service_name: &ServiceName,
+) -> error_stack::Result<u16, ContractError> {
+    let verifier_count: usize = VERIFIERS
+        .prefix(service_name)
+        .range(storage, None, None, Order::Ascending)
+        .filter_map(|v| match v {
+            Ok((_, ver)) => match ver.authorization_state {
+                AuthorizationState::Authorized => Some(()),
+                _ => None,
+            },
+            _ => None,
+        })
+        .collect::<Vec<()>>()
+        .len();
```

Comment:
this copies the entire iterator into a new vector and then takes the length when you're only interested in how many elements you have
```suggestion
        .count();
```